[tohhengxing] iP

tohhengxing opened the pull request:
# A heading (Wojak)

- a
- bullet
- list

1. a
2. numbered
3. list

```java
public class Main {
  // a fenced code block (with syntax highlighting)
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

- [ ] a
- [ ] task
- [ ] list

> a block quote

[a hyper link](https://www.google.com)

an emoji 👎 

`inline.code()`

some text formatting: **bold**, _italic_, ~strikethrough~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

tohhengxing committed:
update: .gitignore

tohhengxing committed:
feat: Rename chatbot

tohhengxing committed:
feat: Add echo

tohhengxing committed:
feat: add List and add to list

tohhengxing committed:
feat: Add Task

tohhengxing committed:
feat: Mark as done

tohhengxing committed:
feat: Refactor tasks

tohhengxing committed:
feat: Major refactor

tohhengxing committed:
fix: Wrong unmark message

tohhengxing committed:
feat: Add todos, events and deadlines

tohhengxing committed:
fix: Minor refactor

tohhengxing committed:
feat: Script testing

tohhengxing committed:
feat: Add errors

tohhengxing committed:
update: Script testing

tohhengxing committed:
feat: Add delete

tohhengxing committed:
feat: Add serialisable and saving

tohhengxing committed:
update: gitignore

tohhengxing committed:
feat: Add deserialiser

tohhengxing committed:
Merge branch 'branch-Level-7'

tohhengxing committed:
feat: Add localdate for Deadline only

tohhengxing committed:
Merge branch 'branch-Level-8'

ckclion commented on src/main/java/controllers/InputParser.java:
```diff
@@ -0,0 +1,132 @@
+        return deleteMatcher.matches();
+    }
+
+    private String extractStringBetweenTwoSubStrings(String command, String prefix, String byMarker) {
```
I like how the method name explains the method

ckclion commented on src/main/java/Wojak.java:
```diff
@@ -0,0 +1,37 @@
+import controllers.errors.InvalidInputError;
+import lib.DbDriverInterface;
+import lib.FileDbDriver;
+import models.*;
```
Perhaps it is better to list imports explicitly

tohhengxing replied:
Listing imports explicitly is useful for context. Importing using wildcard is useful for not writing more code than necessary. I find that only if the project is bigger you'd want to list all your dependencies.

Even then, most IDEs have code completion for Java.

ckclion commented on src/main/java/lib/FileDbDriver.java:
```diff
@@ -0,0 +1,58 @@
+
+    public String init() {
+        try {
+            Path currentWorkingDirectory = Paths.get("").toAbsolutePath();
+            Path parentDirectory = currentWorkingDirectory.getParent();
```
I like that there is good distinction between the directories

ckclion reviewed (commented):
LGTM. Just some nits to fix.

tohhengxing committed:
Merge branch 'add-gradle-support'

tohhengxing committed:
feat: Add gradle

tohhengxing committed:
feat: Add tests

tohhengxing committed:
feat: Add docs for models

tohhengxing committed:
feat: Add docs for controllers

tohhengxing committed:
feat: Add doc for filedbdriver

tohhengxing committed:
feat: Add find

tohhengxing committed:
Merge branch 'branch-Level-9'

tohhengxing committed:
Merge branch 'branch-A-JavaDoc'

ritulkrsingh commented on src/main/java/controllers/InputParser.java:
```diff
@@ -0,0 +1,218 @@
+            String errorString = String.format("%s doesn't exist as a command", cmd);
+            throw new InvalidCommandError(errorString);
+        }
+    }
```
Maybe a switch statement can be used here? (the `equals` methods have to be implemented anyways)

ritulkrsingh commented on src/main/java/controllers/commands/AddDeadlineCommand.java:
I like how there are classes for operation of commands too!

tohhengxing committed:
feat: Add handler, GUI etc

Thanks

tohhengxing committed:
Merge branch 'Level-10'

tohhengxing committed:
feat: Add assert

Add null check assertion in parse method

The parse method does not currently check if the cmd parameter is
null, which could lead to a NullPointerException during execution.

To prevent potential null pointer issues and ensure the command is
never null, add an assertion to verify cmd is not null.

This assertion ensures code correctness and catches errors
early during development, following defensive programming practices.

tohhengxing committed:
refactor: Reduce LOC for parse method

Refactor parse method by delegating command-specific
logic to helper methods

The parse method is too long and violates the
method length limit specified in the course
guidelines.

Long methods are harder to read, maintain, and
understand. Refactoring is required to adhere to
the method length constraint and improve code
modularity.

Extract command-specific logic into separate
helper methods for each command type (mark, delete,
find, unmark, todo, deadline, event).

Breaking down the logic into smaller, focused
methods improves readability, adheres to the 30 LOC
requirement, and follows

tohhengxing committed:
Merge pull request #3 from tohhengxing/branch-A-Assertions

feat: Add assert

tohhengxing committed:
Merge branch 'master' into branch-A-CodeQuality

tohhengxing committed:
Merge pull request #2 from tohhengxing/branch-A-CodeQuality

refactor: Reduce LOC for parse method

tohhengxing committed:
feat: Add changeable db

tohhengxing committed:
feat: Add docs

tohhengxing committed:
fix: Refactor launcher

tohhengxing committed:
update: README

tohhengxing committed:
feat: Update GUI

To conform with requested specificiations

tohhengxing committed:
update: Update UI pic

tohhengxing committed:
fix: Remove bulk imports

tohhengxing committed:
fix: Make task abstract

Because for some reason, it needs to be abstract, when in reality
implementing as an interface makes a lot more sense.

tohhengxing committed:
fix: Enter to fire send button

tohhengxing committed:
fix: Alignment since apparently there's a convention on chatbot design

tohhengxing committed:
feat: Add auto scroll

tohhengxing committed:
update: README.md

tohhengxing committed:
fix: README

tohhengxing committed:
fix: Remove unused variable

tohhengxing committed:
fix: Remove universal import

fix: Update import reorderings
